subprojects/core/src/main/java/org/gradle/internal/buildtree/BuildInclusionCoordinator.java

```diff
@@ -37,12 +42,38 @@
 public class BuildInclusionCoordinator {
     private final Set<IncludedBuildState> loadedBuilds = new CopyOnWriteArraySet<>();
     private final List<IncludedBuildState> libraryBuilds = new CopyOnWriteArrayList<>();
+    private final Map<BuildState, BuildSynchronizer> synchronizers = new ConcurrentHashMap<>();
```

Comment:
🤔I suggest we always use a minimal required data for the data structures. Here, there is no need for the entire `BuildState` to be the key, the lightweight build identity is enough.

Reply:
Whereas I don't have strong opinion on that, I don't believe that it will bring any benefits, so the change is mostly cosmetic. But If your stance on this is strong - I can do that. Although, operating over BuildStates is common for this class

Reply:
`BuildState` works well for this use-case as an identity based map key.